pkg/querier/querier.go

```diff
@@ -109,6 +112,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.MaxSamples, "querier.max-samples", 50e6, "Maximum number of samples a single query can load into memory.")
 	f.DurationVar(&cfg.QueryIngestersWithin, "querier.query-ingesters-within", 0, "Maximum lookback beyond which queries are not sent to ingester. 0 means all queries are sent to ingester.")
 	f.BoolVar(&cfg.EnablePerStepStats, "querier.per-step-stats-enabled", false, "Enable returning samples stats per steps in query response.")
+	f.StringVar(&cfg.PrometheusCodecCompression, "querier.prometheus_codec_compression", "", "Use compression when returning promql response. Supported values 'gzip', 'snappy', and '' (disable compression)")
```

Comment:
We need config validation

Reply:
Can we call it `querier.response-compression`? 

> Use compression when returning promql response

I got that point that the compression is only for promql evaluation APIs. Can we just say it explicitly. `Use compression for metrics query API or instant and range query APIs`?

Reply:
Added validation for both configs